File: news-mobile/lib/function/main_axis_alignment.dart
import 'package:flutter/material.dart';
import 'package:mobile/clients/news_client.dart';

MainAxisAlignment lookupMainAxisAlignment(
  NewsClient client, {
  @required String value,
  MainAxisAlignment fallback = MainAxisAlignment.start,
}) {
  var mainAxisAlignment = fallback;
  if (value == 'start') {
    mainAxisAlignment = MainAxisAlignment.start;
  } else if (value == 'center') {
    mainAxisAlignment = MainAxisAlignment.center;
  } else if (value == 'end') {
    mainAxisAlignment = MainAxisAlignment.end;
  } else if (value == 'spaceEvenly') {
    mainAxisAlignment = MainAxisAlignment.spaceEvenly;
  } else if (value == 'spaceBetween') {
    mainAxisAlignment = MainAxisAlignment.spaceBetween;
  } else if (value == 'spaceAround') {
    mainAxisAlignment = MainAxisAlignment.spaceAround;
  }
  return mainAxisAlignment;
}

File: news-flutter/src/main/java/com/angkorteam/news/flutter/common/MainAxisAlignment.java
package com.angkorteam.news.flutter.common;

public enum MainAxisAlignment {
    start("start"),
    end("end"),
    center("center"),
    spaceBetween("spaceBetween"),
    spaceAround("spaceAround"),
    spaceEvenly("spaceEvenly");

    private String literal;

    MainAxisAlignment(String literal) {
        this.literal = literal;
    }

    public String getLiteral() {
        return literal;
    }

}
